{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidURLResource"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"key"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":", "},{"kind":"externalParam","text":"url"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"URL","preciseIdentifier":"s:10Foundation3URLV"},{"kind":"text","text":")"}],"languages":["swift"],"platforms":["macOS"]}]},{"kind":"content","content":[{"anchor":"discussion","level":2,"type":"heading","text":"Discussion"},{"type":"unorderedList","items":[{"content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"error: The underlying error thrown by file manager."}]}]},{"content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"key: The key used to getting the resource from cache."}]}]},{"content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"url: The disk URL where the target cached file exists."}]}]}]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/symoyanetwork\/symoyanetworkerror\/cacheerrorreason\/invalidurlresource(error:key:url:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork\/SYMoyaNetworkError\/CacheErrorReason\/invalidURLResource(error:key:url:)","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"The file at target URL exists, but its URL resource is unavailable. Code 30003."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidURLResource"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"key"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":", "},{"kind":"externalParam","text":"url"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"URL","preciseIdentifier":"s:10Foundation3URLV"},{"kind":"text","text":")"}],"title":"SYMoyaNetworkError.CacheErrorReason.invalidURLResource(error:key:url:)","roleHeading":"Case","role":"symbol","symbolKind":"case","externalID":"s:13SYMoyaNetwork0aB5ErrorO05CacheC6ReasonO18invalidURLResourceyAEs0C0_p_SS10Foundation3URLVtcAEmF","modules":[{"name":"SYMoyaNetwork"}]},"hierarchy":{"paths":[["doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork","doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork\/SYMoyaNetworkError","doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork\/SYMoyaNetworkError\/CacheErrorReason"]]},"references":{"doc://SYMoyaNetwork/documentation/SYMoyaNetwork/SYMoyaNetworkError":{"role":"symbol","title":"SYMoyaNetworkError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SYMoyaNetworkError"}],"abstract":[{"type":"text","text":"Represents all the errors which can happen in SYMoyaNetwork framework."},{"type":"text","text":" "},{"type":"text","text":"SYMoyaNetwork related methods always throw a "},{"type":"codeVoice","code":"SYMoyaNetworkError"},{"type":"text","text":" or invoke the callback with "},{"type":"codeVoice","code":"SYMoyaNetworkError"},{"type":"text","text":" "},{"type":"text","text":"as its error type. To handle errors from SYMoyaNetwork, you switch over the error to get a reason catalog,"},{"type":"text","text":" "},{"type":"text","text":"then switch over the reason to know error detail."}],"identifier":"doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork\/SYMoyaNetworkError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"SYMoyaNetworkError"}],"url":"\/documentation\/symoyanetwork\/symoyanetworkerror"},"doc://SYMoyaNetwork/documentation/SYMoyaNetwork/SYMoyaNetworkError/CacheErrorReason/invalidURLResource(error:key:url:)":{"role":"symbol","title":"SYMoyaNetworkError.CacheErrorReason.invalidURLResource(error:key:url:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidURLResource"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"key"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":", "},{"kind":"externalParam","text":"url"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"URL","preciseIdentifier":"s:10Foundation3URLV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"The file at target URL exists, but its URL resource is unavailable. Code 30003."}],"identifier":"doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork\/SYMoyaNetworkError\/CacheErrorReason\/invalidURLResource(error:key:url:)","kind":"symbol","type":"topic","url":"\/documentation\/symoyanetwork\/symoyanetworkerror\/cacheerrorreason\/invalidurlresource(error:key:url:)"},"doc://SYMoyaNetwork/documentation/SYMoyaNetwork":{"role":"collection","title":"SYMoyaNetwork","abstract":[{"type":"text","text":"Based on the network abstraction of secondary encapsulation of "},{"type":"codeVoice","code":"Moya"},{"type":"text","text":". Maintain the same usage method of "},{"type":"codeVoice","code":"Moya"},{"type":"text","text":"â€™s "},{"type":"codeVoice","code":"TargetType"},{"type":"text","text":" implements commonly used data parsing, supporting: "},{"type":"codeVoice","code":"HandyJSON"},{"type":"text","text":","},{"type":"codeVoice","code":"ObjectMapper"},{"type":"text","text":", "},{"type":"codeVoice","code":"Codable"},{"type":"text","text":", "},{"type":"codeVoice","code":"SwiftyJSON"},{"type":"text","text":", Developer there is no need to write boilerplate code for data parsing. You only need to care about what kind of data you want and select the data type. "},{"type":"codeVoice","code":"SYMoyaNetwork"},{"type":"text","text":" has done all this. "},{"type":"codeVoice","code":"SYMoyaNetwork"},{"type":"text","text":" also implements network caching, and configure common caching strategies to only implement the desired caching strategy. "},{"type":"codeVoice","code":"Response"},{"type":"text","text":" will perform cache synchronization according to the policy. Developers no longer need to spend a lot of time doing such work. "},{"type":"codeVoice","code":"SYMoyaNetwork"},{"type":"text","text":" allows data requests Become simple, less developers write boilerplate code, more time to focus on business."}],"identifier":"doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork","kind":"symbol","type":"topic","url":"\/documentation\/symoyanetwork"},"doc://SYMoyaNetwork/documentation/SYMoyaNetwork/SYMoyaNetworkError/CacheErrorReason":{"role":"symbol","title":"SYMoyaNetworkError.CacheErrorReason","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"CacheErrorReason"}],"abstract":[{"type":"text","text":"Represents the error reason during SYMoyaNetwork caching system."}],"identifier":"doc:\/\/SYMoyaNetwork\/documentation\/SYMoyaNetwork\/SYMoyaNetworkError\/CacheErrorReason","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"CacheErrorReason"}],"url":"\/documentation\/symoyanetwork\/symoyanetworkerror\/cacheerrorreason"}}}